external/fv3fit/fv3fit/reservoir/domain.py

```diff
@@ -202,7 +209,7 @@ def load(cls, path):
             metadata = yaml.safe_load(f)
         return cls(**metadata)
 
-    def merge_subdomains(self, flat_prediction: np.ndarray):
+    def merge_subdomains(self, flat_prediction: np.ndarray) -> np.ndarray:
```

Comment:
The initial comment in this method kind of alludes to this, but I would add a small docstring description that this rebuilds flattened readout individual subdomains back into the original x, y, z layout.  `merge_subdomains` makes sense in context of comments and the plural use of `subdomains` but it can be a bit misleading at the higher level since `unstack_subdomain` also exists and the difference is subtle.

Reply:
It also seems like `unstack_subdomain` method could be used somewhere here with the similarity in calculating the vertical dimension size, but there is the slight difference of adding the subdomain size and the overlap always being `False`.  Nothing required for this comment.  Just making a public note of it.